### PR TITLE
Last minute improvements/fixes to substructure exercise

### PR DESCRIPTION
**Pull request by DryRun**
- Add W(qq) HT-binned sample, so we have more than just QCD vs top everywhere.
   - Add truth matching to FWlite script; otherwise, W is obscured by extra jets.
- Use RSG KK gluon (3 TeV) for most boosted top plots, so the tops are fully merged usually.
- Improved plotting: better colors, ability to specify which processes to plot.

### CODE REVIEW

File: scripts/jmedas_make_histograms.py [hunk 1]
```diff
@@ -185,7 +185,7 @@
 else:
     doMatchingAK8 = False
 
-if doMatchingAK4 or doMatchingAK8:
```
Comment (kirschen) [inline scripts/jmedas_make_histograms.py]: Do you really need to override this?

File: scripts/jmedas_make_histograms.py [hunk 2]
```diff
@@ -415,10 +419,11 @@ def getJER(jetEta, sysType) :
         rhoValue = rhoHandle.product()
         pvs = pvHandle.product()
         #print rhoValue[0]
+
+        event.getByLabel(genParticlesLabel, genParticlesHandle)
+        genParticles = genParticlesHandle.product()
         
         if doMatchingAK4 or doMatchingAK8:
-            event.getByLabel(genParticlesLabel, genParticlesHandle)
-            genParticles = genParticlesHandle.product()
```
Comment (kirschen) [inline scripts/jmedas_make_histograms.py]: Do you need the genparticles even when not doing the matching?

Comment (kirschen) [reply scripts/jmedas_make_histograms.py]: Suggest to revert

File: scripts/jmedas_make_histograms.py [hunk 3]
```diff
@@ -588,6 +593,13 @@ def getJER(jetEta, sysType) :
         # loop over jets and fill hists
         if args.verbose :
           print jets1.size()
+
+        # DEBUG : Print gen info
+        #print("\n\n*** RSG particles ***")
+        #for genPart in genParticles:
+        #  if abs(genPart.pdgId()) > 1000 or abs(genPart.pdgId()) == 6 or abs(genPart.pdgId()) == 24:
+        #    print("{} {} {} {}".format(genPart.pdgId(), genPart.status(), genPart.mass(), genPart.pt()))
+
```
Comment (kirschen) [inline scripts/jmedas_make_histograms.py:602]: remove commented debug again?

**Review by kirschen — APPROVED** [review]
Thanks, no objections on merging, but would suggest some remnant edits from debugging (always read in genparticles)

**Comment by kirschen** [comment]
Can you merge yourself after the edits? Thanks!

File: scripts/jmedas_make_histograms.py (hunk 1 above)
Comment (kirschen) [reply scripts/jmedas_make_histograms.py]: not sure what I was commenting on, here, some intermediate commit? Never mind... resolved

File: scripts/jmedas_make_histograms.py [hunk 4]
```diff
@@ -159,6 +169,25 @@
...
+if args.matchPdgIdAK8:
+    print "Doing AK8 truth matching."
+    doMatchingAK8 = True
+    matchAK8PdgId = int(args.matchPdgIdAK8[0])
+    matchAK8DR = float(args.matchPdgIdAK8[1])
+else:
+    doMatchingAK8 = False
+
+if doMatchingAK4 or doMatchingAK8 or True:
```
Comment (kirschen) [inline scripts/jmedas_make_histograms.py:188]: Why the True here?

**Review by kirschen — APPROVED** [review]
Looks good, just check the two remaning comments before merging, please 👍

**Comment by DryRun** [comment]
Thanks @kirschen , I've incorporated your review feedback.

**Pull request merged**